packages/config/types/router.d.ts

```diff
@@ -6,9 +6,14 @@
  */
 
 import { RouterOptions, RouteConfig } from 'vue-router'
+import Vue, { ComponentOptions, AsyncComponent } from 'vue'
+
+type Component = ComponentOptions<Vue> | typeof Vue | AsyncComponent;
 
 export interface NuxtRouteConfig extends RouteConfig {
-  chunkNames: { [key: string]: string }
+  chunkNames: { [key: string]: string },
+  component: Component | string,
```

Comment:
```suggestion
  component: RouteConfig['component'] | string
```

It will directly infer RouteConfig.component type (which is Component)

Reply:
Good to know that, thank you